packages/bs-neo-legacy/src/DoraBDSNeoLegacy.ts

```diff
@@ -8,72 +8,79 @@ import {
+        return {
+          amount: Number(transfer.value),
           from: array[array.length - 1]?.address,
-          hash: transfer.asset,
+          contractHash: transfer.asset,
           to: transfer.address,
-          type: 'asset',
-        }))
-      : []
+          type: 'token',
+          token,
+        }
+      })
+    )
 
     return {
       hash: data.txid,
       block: data.block,
-      netfee: data.net_fee ?? '0',
-      sysfee: data.sys_fee ?? '0',
-      totfee: (Number(data.sys_fee) + Number(data.net_fee)).toString(),
-      time: data.time,
+      fee: Number(data.sys_fee ?? 0) + Number(data.net_fee ?? 0),
+      time: Number(data.time),
       notifications: [], //neoLegacy doesn't have notifications
       transfers,
     }
   }
 
-  async getHistoryTransactions(address: string, page: number = 1): Promise<TransactionHistoryResponse> {
-    const data = await api.NeoLegacyREST.getAddressAbstracts(address, page, this.network.type)
+  async getTransactionsByAddress(address: string, page: number = 1): Promise<TransactionHistoryResponse> {
+    const data = await api.NeoLegacyREST.getAddressAbstracts(address, page, this.networkType)
     const transactions = new Map<string, TransactionResponse>()
 
-    data.entries.forEach(entry => {
-      if (entry.address_from !== address && entry.address_to !== address) return
-
-      const transfer: TransactionTransferAsset = {
-        amount: entry.amount.toString(),
-        from: entry.address_from ?? 'Mint',
-        to: entry.address_to ?? 'Burn',
-        type: 'asset',
-      }
-
-      const existingTransaction = transactions.get(entry.txid)
-      if (existingTransaction) {
-        existingTransaction.transfers.push(transfer)
-        return
-      }
-
-      transactions.set(entry.txid, {
-        block: entry.block_height,
-        hash: entry.txid,
-        netfee: '0',
-        sysfee: '0',
-        totfee: '0',
-        time: entry.time.toString(),
-        transfers: [transfer],
-        notifications: [],
+    await Promise.all(
+      data.entries.map(async entry => {
+        if (entry.address_from !== address && entry.address_to !== address) return
+
+        const token = await this.getTokenInfo(entry.asset)
+        const transfer: TransactionTransferAsset = {
+          amount: Number(entry.amount),
+          from: entry.address_from ?? 'Mint',
+          to: entry.address_to ?? 'Burn',
+          type: 'token',
+          contractHash: entry.asset,
+          token,
+        }
+        const existingTransaction = transactions.get(entry.txid)
+        if (existingTransaction) {
+          existingTransaction.transfers.push(transfer)
+          return
+        }
+
+        transactions.set(entry.txid, {
+          block: entry.block_height,
+          hash: entry.txid,
+          time: entry.time,
+          transfers: [transfer],
+          notifications: [],
+        })
       })
-    })
+    )
```

Comment:
I think this would be less weird with a for of loop:
```javascript
for (const { address_from, address_to, asset, amount, block_height, time, txid } of data.entries) {
  if (address_from !== address && address_to !== address) continue;

  const token = await this.getTokenInfo(asset);
  const transfer: TransactionTransferAsset = {
    amount: Number(amount),
    from: address_from ?? 'Mint',
    to: address_to ?? 'Burn',
    type: 'token',
    contractHash: asset,
    token,
  };
  const existingTransaction = transactions.get(txid);
  if (existingTransaction) {
    existingTransaction.transfers.push(transfer);
    continue;
  }

  transactions.set(txid, {
    block: block_height,
    hash: txid,
    time,
    transfers: [transfer],
    notifications: [],
  });
}
```

Reply:
Promise.all executes all promises in parallel, while the for-loop would wait each one individually.
But I agree this syntax can be improved by creating a method that returns the promise and use them on the Promise.all